src/frontend/src/pages/HomePage/UpcomingDeadlines.tsx

```diff
@@ -22,6 +22,12 @@ import LoadingIndicator from '../../components/LoadingIndicator';
 import PageBlock from '../../layouts/PageBlock';
 import ErrorPage from '../ErrorPage';
 import { Grid, Typography, useTheme } from '@mui/material';
+import DetailDisplay from '../../components/DetailDisplay';
+import { DetailDisplayProps } from '../../components/DetailDisplay';
+
+const UpcomingDeadlinesDetailDisplay: React.FC<DetailDisplayProps> = ({ label, content }) => {
+  return <DetailDisplay label={label} content={content} paddingRight={2}></DetailDisplay>;
+};
```

Comment:
Making a whole component within the file is a bit overkill, I would just use detail display in all these places